### PR TITLE
dependency Update to fix Nodev18

### PR DESCRIPTION
**Pull request by thomasaugsten**

### CODE REVIEW

**Comment by MikeMcC399** [comment]
Updating autoprefixer causes the warning:

`autoprefixer: \cwa-website\src\assets\scss\style.css:4753:3: Replace color-adjust to print-color-adjust. The color-adjust shorthand is currently deprecated.
`

File: package.json [hunk 1]
```diff
@@ -52,30 +52,30 @@
...
     "rxjs": "^7.5.7",
     "sass": "^1.55.0",
-    "screenfull": "^5.2.0",
+    "screenfull": "^6.0.2",
     "slick-carousel": "^1.8.1",
     "uint8-to-base64": "^0.2.0",
     "vinyl-named": "^1.1.0",
-    "webpack": "^4.46.0",
-    "webpack-stream": "^5.2.1",
-    "yargs": "^15.4.1"
```
Comment (MikeMcC399) [inline package.json:70]: Please revert to "yargs": "^15.4.1"

**Review by MikeMcC399 — CHANGES_REQUESTED** [review]
Please revert to "yargs": "^15.4.1"

Updating to "yargs": "^17.6.0" causes the pending PR https://github.com/corona-warn-app/cwa-website/pull/3181 to fail. It seems this is due to a change in yargs in v16 which supports ESM.

`npm run build` with "yargs": "^15.4.1" and Node.js v18.12.0 runs successfully, so it is not necessary to update this package in order get Node.js 18 compatibility. 

An upgrade of `yargs` to v17 could be looked at in a separate PR which may then also need changes to `gulpfile`.
Edit: see solution in https://github.com/corona-warn-app/cwa-website/pull/3181#issuecomment-1295191701.

**Review by MikeMcC399 — APPROVED** [review]
Thank you for the reversion of yargs back to its original v15! That solves the incompatibility with PR https://github.com/corona-warn-app/cwa-website/pull/3181.

**Comment by brianebeling** [comment]
Hey, I tested this with the following steps:

- Clone PR
- Run `npm ci` for a clean install / or delete node_modules and run `npm i`
- `npx cypress run -s 'cypress/integration/*.js' -c baseUrl=https://coronawarn.app --headless --browser chrome`
- `npm start` and click through the website, check for errors, test functionality

All seems to work fine. Please let us know if you find anything not working as expected and/or create a new issue.

The following deprecation warnings are new or should be looked at:

```bash
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated source-map-resolve@0.6.0: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
npm WARN deprecated highlight.js@8.9.1: Version no longer supported. Upgrade to @latest
```

**Comment by MikeMcC399** [comment]
@brianebeling 

The deprecation notices look very similar to the ones which appear when running `npm ci` under node v16.18.0 on the current `master` branch.

```
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
npm WARN deprecated highlight.js@8.9.1: Version no longer supported. Upgrade to @latest
```

**Comment by MikeMcC399** [comment]
When I tested on the current `master` branch with node v18.12.0 running `npm ci` followed by `npm run build`, the build gave the error 
`Error: error:0308010C:digital envelope routines::unsupported`
This was fixed by
`npm install webpack@latest` which updated from v4 to v5.

I didn't find it necessary to update any of the other npm packages in order to get the web to build and run under Node.js 18.

So the other package updates, apart from `webpack` in this PR aren't necessary to reach Node.js 18 compatibility. It may still be good to have them updated if there will still be much further development in this repo.

**Comment by MikeMcC399** [comment]
Successfully tested on Ubuntu 20.04.5 LTS with Google Chrome 107:

- Cherry-picked f2018a26 from PR https://github.com/corona-warn-app/cwa-website/pull/3185 to avoid check_link.js error

```
npm ci
npm test
```

with `node -v` 
- v16.18.0 and 
- v18.12.0

Successfully tested on Ubuntu 20.04.5 LTS with Google Chrome 107:

(Cherry-picked f2018a26 from PR https://github.com/corona-warn-app/cwa-website/pull/3185)

```
npm ci
npm test
```

with `node -v`
- v16.18.0 and 
- v18.12.0

```
       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  app_to_web.js                            00:21        8        8        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  applink_spec.js                          00:01        3        3        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  blog_spec.js                             00:01        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  check_anchor_links.js                    01:54       37       37        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  check_videos.js                          00:57       39       39        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  eventRegistration.js                     00:12        6        6        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  faq_link_attr.js                         00:06        3        3        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  faq.js                                   00:18       16       16        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  hotline_spec.js                          938ms        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  hybrid/check_links.js                    15:55       39       39        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  mime.js                                  720ms        2        2        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        19:52      155      155        -        -        -  
```

**Comment by MikeMcC399** [comment]
- I have submitted an alternative PR https://github.com/corona-warn-app/cwa-website/pull/3187 which migrates the production web, including build and deploy, to Node.js 18 using the minimum amount of changes.
- Please consider if this is a good alternative way to approach the migration.

**Comment by MikeMcC399** [comment]
If the branch [depUpdateNode18](https://github.com/corona-warn-app/cwa-website/tree/depUpdateNode18) is no longer needed, which is typically the case after a PR has been merged, then it should be deleted.

**Pull request merged**